services/graphql-server/src/graphql/utils/retrieve-youtube-videos.js

```diff
@@ -0,0 +1,35 @@
+const { get } = require('@base-cms/object-path');
+const client = require('../../google-data-api-client');
+
+const retrievePlaylistId = async ({ _id, youtube }, basedb) => {
+  const playlistId = get(youtube, 'playlistId');
+  if (playlistId) return playlistId;
+
+  const id = get(youtube, 'channelId');
+  const forUsername = get(youtube, 'username');
+  if (id || forUsername) {
+    const payload = {
+      part: 'contentDetails',
+      ...(id && { id }),
+      ...(forUsername && { forUsername }),
+    };
+    const response = await client.request('youtube.channelList', payload);
+    const value = get(response, 'items.0.contentDetails.relatedPlaylists.uploads');
+    if (value) basedb.updateOne('platform.Content', { _id }, { $set: { 'youtube.playlistId': value } });
```

Comment:
This is generally a no-no -- you're running a query that's also mutating data. Is there another way to accomplish this?